applications/tari_base_node/src/main.rs

```diff
@@ -213,12 +246,34 @@ fn main_inner() -> Result<(), ExitCodes> {
         "Node has been successfully configured and initialized. Starting CLI loop."
     );
 
-    cli_loop(parser, shutdown);
+    if !bootstrap.rebuild_db {
```

Comment:
This is pretty messy since you do another check a few lines later.
and unscalable (what if there's a `bootstrap.foo` option in future?)

Reply:
All these issues go away if recovery is its own separate module